src/metabase/api/table.clj

```diff
@@ -63,6 +69,34 @@
         (t2/hydrate :db :pk_field)
         fix-schema)))
 
+(api.macros/defendpoint :get "/:table-id/data"
+  "Get the data for the given table"
+  [{:keys [table-id]} :- [:map [:table-id ms/PositiveInt]]]
+  (let [table (t2/select-one :model/Table :id table-id)
+        db-id (:db_id table)]
+    (api/read-check table)
+    (qp.store/with-metadata-provider db-id
+      (let [mp       (qp.store/metadata-provider)
+            query    (-> (lib/query mp (lib.metadata/table mp table-id))
+                         (update-in [:middleware :js-int-to-string?] (fnil identity true))
+                         qp/userland-query-with-default-constraints
+                         (update :info merge {:executed-by api/*current-user-id*
+                                              :context     :table-grid
+                                              :card-id     nil}))]
+        (events/publish-event! :event/table-read {:object  (t2/select-one :model/Table :id table-id)
```

Comment:
why `select-one` here is you have `table` in `let` already?

Reply:
Fixed 😄